extensions/amp-auto-ads/0.1/placement.js

```diff
@@ -181,16 +182,23 @@ export class Placement {
           return this.state_;
         }
         this.adElement_ = this.createAdElement_(baseAttributes, sizing.width);
-        this.injector_(this.anchorElement_, this.adElement_);
-        return this.resources_.attemptChangeSize(this.adElement_,
-            sizing.height || TARGET_AD_HEIGHT_PX,
-            undefined, this.margins_)
+        this.injector_(this.anchorElement_, this.getAdElement());
+
+        // CustomElement polyfill does not call connectedCallback synchronously.
+        // So we explicitly wait for CustomElement to be ready.
+        return whenUpgradedToCustomElement(this.getAdElement())
+            .then(() => this.getAdElement().whenBuilt())
             .then(() => {
-              this.state_ = PlacementState.PLACED;
-              return this.state_;
-            }, () => {
-              this.state_ = PlacementState.RESIZE_FAILED;
-              return this.state_;
+              return this.resources_.attemptChangeSize(this.getAdElement(),
```

Comment:
What is the size of the ad before this runs?

Reply:
all ads are inserted with 0 height to prevent a jump.